src/FreeAge/common/object_types.hpp

```diff
@@ -16,8 +16,12 @@ enum class ObjectType {
 constexpr u32 kInvalidObjectId = std::numeric_limits<u32>::max();
 
 enum class InteractionType {
+  // A pseudo intercation type that describes that intercation is not specified and 
+  // must be determined by a call to GetInteractionType. To not be confused with ::Invalid.
+  Unknown = 0,
```

Comment:
I side change I did was to store the interaction in the unit. You where using the `GetInteractionType` to find the interaction. The problem was for example the `DropoffResources`, `Repair` and `Garrison` could all be possibilities for a single scenario. 

I changed it so now the `SetUnitTarget` also the takes an optional** `InteractionType` and uses that instead of the calculating by `GetInteractionType`. If no intercation is passed the goes back to the old way and uses the `GetInteractionType`.

** `InteractionType::Unknown` is how implement the "optional".

Reply:
Sounds good. What do you think about naming this `InteractionType::Default` instead of `InteractionType::Unknown`? For me that would make it clearer what this does.